src/K.hs

```diff
@@ -146,6 +146,10 @@ kExprInt (Div a b) = "(" <> kExprInt a <> " /Int " <> kExprInt b <> ")"
 kExprInt (Mod a b) = "(" <> kExprInt a <> " modInt " <> kExprInt b <> ")"
 kExprInt (Exp a b) = "(" <> kExprInt a <> " ^Int " <> kExprInt b <> ")"
 kExprInt (LitInt a) = show a
+kExprInt (IntMin a) = kExprInt $ LitInt $ 0 - 2 ^ (a - 1)
+kExprInt (IntMax a) = kExprInt $ LitInt $ 2 ^ (a - 1) - 1
+kExprInt (UIntMin a) = kExprInt $ LitInt 0
+kExprInt (UIntMax a) = kExprInt $ LitInt $ 2 ^ a - 1
```

Comment:
I think this is all that's needed for K, but I don't really understand the SMT implementation. @MrChico @xwvvvvwx can you help?